app/[lang]/dictionaries.ts

```diff
@@ -0,0 +1,21 @@
+import 'server-only';
+
+type Obj<T> = {
+  [k: string]: T;
+};
+
+interface DictionaryInterface {
+  [k: string]: () => Promise<Obj<Obj<string>>>;
```

Comment:
better to use inferred types here. so you will get completions on the items in dictionary